src/Features/Blockcore.Features.ColdStaking/ColdStakingManager.cs

```diff
@@ -344,7 +346,16 @@ internal HdAddress GetFirstUnusedColdStakingAddress(string walletName, bool isCo
 
             // Get/create the cold staking accounts.
             HdAccount coldAccount = this.GetOrCreateColdStakingAccount(walletName, true, walletPassword);
-            HdAccount hotAccount = this.GetOrCreateColdStakingAccount(walletName, false, walletPassword);
+            HdAccount hotAccount;
+
+            if (createHotAccount)
```

Comment:
Wait the problem is only with hot accounts?

Reply:
The problem is having both active at the same time, there is no need for the reversed.